ai_ta_backend/vector_database.py

```diff
@@ -992,15 +1051,13 @@ def context_padding(self, found_docs, search_query, course_name):
     for doc in found_docs: # top N from QDRANT

```

Comment:
Really need to parallelize this loop. gpt-4 can probably do that for you with asyncio.